h/services/list_groups.py

```diff
@@ -67,17 +68,52 @@ def open_groups(self, user=None, authority=None, document_uri=None):
                       .all())
         return self._sort(groups)
 
-    def private_groups(self, user=None):
+    def _private_groups(self, user=None):
         """Return this user's private groups per user.groups."""
 
         if user is None:
             return []
         return self._sort(user.groups)
 
+    def _parse_origin(self, uri):
```

Comment:
I'd especially like to get eyes on this method to see if I did this  ham-fistedly or in ways that might not be compatible or good.

Reply:
I should also point out that neither this nor the `_world_group` method are yet called by anything in the service.

Reply:
This looks fine to me, and I think it's well tested by your parametrized tests so that gives me confidence in it. I like that we're using `SplitResult.geturl()` instead of formatting our own string with `://` hardcoded